### PR TITLE
[MTDSA-10540] Create Foreign Property Period Summary V2 Validator

### PR DESCRIPTION
**Pull request by Mbrookesy**

### CODE REVIEW

File: test/v2/controllers/requestParsers/validators/CreateForeignPropertyPeriodSummaryValidatorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,2085 @@
...
+            |        "other": 2425.11
+            |      }
+            |    }
+            |  ]
+            |}
+            |""".stripMargin
+        ))) shouldBe List(CountryCodeFormatError.copy(paths = Some(Seq("/foreignNonFhlProperty/0/countryCode", "/foreignNonFhlProperty/1/countryCode"))))
+      }
+    }
+
```
Comment (jeremystone) [inline test/v2/controllers/requestParsers/validators/CreateForeignPropertyPeriodSummaryValidatorSpec.scala:1932]: Add tests for duplicate country code detection (recently added to the spec). See [AmendForeignPropertyPeriodSummaryValidator#L71](https://github.com/hmrc/property-business-api/blob/d645b7f47057179151e176825b7195583ece0e46/app/v2/controllers/requestParsers/validators/AmendForeignPropertyPeriodSummaryValidator.scala#L71) and [AmendForeignPropertyPeriodSummaryValidatorSpec#L429](https://github.com/hmrc/property-business-api/blob/d645b7f47057179151e176825b7195583ece0e46/test/v2/controllers/requestParsers/validators/AmendForeignPropertyPeriodSummaryValidatorSpec.scala#L429) (both currently under review :grimacing:)

File: test/v2/controllers/requestParsers/CreateForeignPropertyPeriodSummaryRequestParserSpec.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      |      "income":{
+      |         "rentAmount":381.21
+      |      }
+      |   }
+      |}
+      |""".stripMargin
+  )
+
+  val inputData: CreateForeignPropertyPeriodSummaryRawData =
+    CreateForeignPropertyPeriodSummaryRawData(nino, taxYear, businessId, requestBodyJson)
```
Comment (jeremystone) [inline test/v2/controllers/requestParsers/CreateForeignPropertyPeriodSummaryRequestParserSpec.scala:50]: Use named parameters and check correct - it will highlight a problem with the parser code where the ordering is also wrong.
